osqp_mex.cpp

```diff
@@ -464,196 +372,134 @@ void mexFunction(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[])
-        }
-
-        // Fill x and y
-        const mxArray *x = prhs[2];
-        const mxArray *y = prhs[3];
-
-
-        // Copy vectors to ensure they are cast as c_float
-        c_float *x_vec;
-        c_float *y_vec;
-
-        if(!mxIsEmpty(x)){
-            x_vec = copyToCfloatVector(mxGetPr(x),
-                                       osqpData->work->data->n);
-        }
-        if(!mxIsEmpty(y)){
-            y_vec = copyToCfloatVector(mxGetPr(y),
-                                       osqpData->work->data->m);
+                mexErrMsgTxt("Matrix update error!");
         }
 
-        // Warm start x and y
-        osqp_warm_start(osqpData->work, x_vec, y_vec);
-
-        // Free vectors
-        if(!mxIsEmpty(x)) c_free(x_vec);
-        if(!mxIsEmpty(y)) c_free(y_vec);
-
         return;
     }
 
-
-    // Warm start x variable
-    if (!strcmp("warm_start_x", cmd)) {
-
-        //throw an error if this is called before solver is configured
-        if(!osqpData->work){
-          mexErrMsgTxt("Solver has not been initialized.");
-        }
-
-        // Fill x and y
-        const mxArray *x = prhs[2];
-
-
-        // Copy vectors to ensure they are cast as c_float
-        c_float *x_vec;
-
-        if(!mxIsEmpty(x)){
-            x_vec = copyToCfloatVector(mxGetPr(x),
-                                       osqpData->work->data->n);
-        }
-
-        // Warm start x
-        osqp_warm_start_x(osqpData->work, x_vec);
-
-        // Free vectors
-        if(!mxIsEmpty(x)) c_free(x_vec);
-
-        return;
-    }
-
-
-    // Warm start y variable
-    if (!strcmp("warm_start_y", cmd)) {
-
-        //throw an error if this is called before solver is configured
-        if(!osqpData->work){
+    if (!strcmp("warm_start", cmd) || !strcmp("warm_start_x", cmd) || !strcmp("warm_start_y", cmd)) {
+      
+      //throw an error if this is called before solver is configured
+      if(!osqpData->solver){
           mexErrMsgTxt("Solver has not been initialized.");
         }
 
-        // Fill x and y
-        const mxArray *y = prhs[2];
+      // Fill x and y
+      const mxArray *x = NULL;
+      const mxArray *y = NULL;
+      if (!strcmp("warm_start", cmd)) {  
+        x = prhs[2];
+        y = prhs[3];
+      }
+      else if (!strcmp("warm_start_x", cmd)) {
+        x = prhs[2];
+        y = NULL;
+      }
 
+      else if (!strcmp("warm_start_y", cmd)) {
+        x = NULL;
+        y = prhs[2];
+      }
 
-        // Copy vectors to ensure they are cast as c_float
-        c_float *y_vec;
+      // Copy vectors to ensure they are cast as OSQPFloat
+      OSQPFloat *x_vec = NULL;
+      OSQPFloat *y_vec = NULL;
 
-        if(!mxIsEmpty(y)){
-            y_vec = copyToCfloatVector(mxGetPr(y),
-                                       osqpData->work->data->m);
-        }
+      if(!mxIsEmpty(x)){
+          x_vec = copyToOSQPFloatVector(mxGetPr(x),
+                                      osqpData->solver->work->data->n);

```

Comment:
Same here (don't reach into the work struct, use the public dimensions function).

Reply:
Will be changed in the next push